src/me/john000708/barrels/Barrel.java

```diff
@@ -82,16 +86,35 @@ public boolean canOpen(Block b, Player p) {
             }
 
             @Override
-            public int[] getSlotsAccessedByItemTransport(ItemTransportFlow itemTransportFlow) {
+            public int[] getSlotsAccessedByItemTransport(ItemTransportFlow flow) {
                 return new int[0];
             }
 
             @Override
-            public int[] getSlotsAccessedByItemTransport(BlockMenu menu, ItemTransportFlow flow, ItemStack item) {
+            public int[] getSlotsAccessedByItemTransport(DirtyChestMenu menu, ItemTransportFlow flow, ItemStack item) {
                 if (flow == ItemTransportFlow.INSERT) {
-                    if (BlockStorage.getLocationInfo(menu.getLocation(), "storedItems") != null)
-                        return isSimilar(item, menu.getItemInSlot(22)) ? getInputSlots() : new int[0];
-                    else return getInputSlots();
+
+                    List<Integer> slots = new ArrayList<>();
+
+                    for (int slot : getInputSlots()) {
+                        if (SlimefunManager.isItemSimilar(menu.getItemInSlot(slot), item, true)) {
+                            slots.add(slot);
+                        }
+                    }
+                    if (slots.isEmpty()) {
+                        return getInputSlots();
+                    }
+                    else {
+                        Collections.sort(slots, compareSlots(menu));
+                        int[] array = new int[slots.size()];
+
+                        for (int i = 0; i < slots.size(); i++) {
+                            array[i] = slots.get(i);
+                        }
+
+                        return array;
+                    }
+
```

Comment:
What the fuck are you doing here???

Why are you replacing all of this with something completely different?

Reply:
The replacement allowed for the cargo net to be able to input and output items from the barrel, whereas for the original set of code for it, it worked for older versions, but it did not work now. I do, however, now see the issue as to why this can cause other errors, because it apparently does not take out the items from the output side of the barrel, rather the input side (Tested a moment ago).